src/jit/lclvars.cpp

```diff
@@ -1963,6 +1981,25 @@ void   Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, boo
             varDsc->lvBaseType = simdBaseType;
         }
 #endif // FEATURE_SIMD
+#ifdef FEATURE_HFA
```

Comment:
A Vector2/3/4 type is both a HFA and also a SIMD type on Arm64.  Like the logic here, we have to make sure that FEATURE_SIMD and FEATURE_HFA co-exist and play nice with each other.